tests/annotations/tests.py

```diff
@@ -619,3 +619,16 @@ def test_annotation_filter_with_subquery(self):
             total_books=Subquery(long_books_qs, output_field=IntegerField()),
         ).values('name')
         self.assertCountEqual(publisher_books_qs, [{'name': 'Sams'}, {'name': 'Morgan Kaufmann'}])
+
+    def test_annotation_filter_with_subquery_name_colission(self):
```

Comment:
Given we're testing `GROUP BY` logic the test should likely live in aggregation tests instead. There's likely an existing model you can reuse there.